pkg/controller/master/backup/restore.go

```diff
@@ -695,75 +551,115 @@ func (h *RestoreHandler) createRestoredPVC(vmRestore *harvesterv1.VirtualMachine
 	return err
 }
 
-func getRestoreID(vmRestore *harvesterv1.VirtualMachineRestore) string {
-	return fmt.Sprintf("%s-%s", vmRestore.Name, vmRestore.UID)
-}
-
-func (h *RestoreHandler) doUpdateError(original, updated *harvesterv1.VirtualMachineRestore, err error, createEvent bool) error {
-	if createEvent {
-		h.recorder.Eventf(
-			updated,
-			corev1.EventTypeWarning,
-			restoreErrorEvent,
-			"VirtualMachineRestore encountered error %s",
-			err.Error(),
-		)
+func (h *RestoreHandler) deleteOldPVC(vmRestore *harvesterv1.VirtualMachineRestore, vm *kv1.VirtualMachine) error {
+	if isNewVMOrHasRetainPolicy(vmRestore) {
+		logrus.Infof("skip deleting old PVC of vm %s/%s", vm.Name, vm.Namespace)
+		return nil
 	}
-	updateRestoreCondition(updated, newProgressingCondition(corev1.ConditionFalse, err.Error()))
-	updateRestoreCondition(updated, newReadyCondition(corev1.ConditionFalse, err.Error()))
-	if err2 := h.doUpdate(original, updated); err2 != nil {
-		return err2
+
+	// clean up existing pvc
+	for _, volName := range vmRestore.Status.DeletedVolumes {
+		vol, err := h.pvcCache.Get(vmRestore.Namespace, volName)
+		if err != nil && !apierrors.IsNotFound(err) {
+			return err
+		}
+
+		if vol != nil {
+			err = h.pvcClient.Delete(vol.Namespace, vol.Name, &metav1.DeleteOptions{})
+			if err != nil {
+				return err
+			}
+		}
 	}
 
-	return err
+	return nil
 }
 
-func updateRestoreCondition(r *harvesterv1.VirtualMachineRestore, c harvesterv1.Condition) {
-	r.Status.Conditions = updateCondition(r.Status.Conditions, c, true)
+func (h *RestoreHandler) restartVM(vm *kv1.VirtualMachine) error {
+	logrus.Infof("restarting the vm %s, current state running:%v", vm.Name, *vm.Spec.Running)
+	var running = true
+	if vm.Spec.Running == nil || vm.Spec.Running == &running {
```

Comment:
`*vm.Spec.Running` instead of `vm.Spec.Running == &running`

Reply:
Fixed it. Thank you.